new-packages/ts-project/__tests__/state.test.ts

```diff
@@ -0,0 +1,695 @@
+                    },
+                    "id": "state1.state2",
+                    "states": [
+                      {
+                        "data": {
+                          "description": undefined,
+                          "entry": [],
+                          "exit": [],
+                          "history": undefined,
+                          "initial": undefined,
+                          "invoke": [],
+                          "metaEntries": [],
+                          "tags": [],
+                          "type": undefined,
+                        },
+                        "id": "state1.state2.state3",
+                        "states": [],
+                      },
+                    ],
+                  },
+                ],
+              },
+              {
+                "data": {
+                  "description": undefined,
+                  "entry": [],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": undefined,
+                },
+                "id": "state4",
+                "states": [],
+              },
+            ],
+          },
+          "states": [],
+        },
+        [],
+      ],
+    ]
+  `);
+});
+
+test('Should extract state.initial with string value', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        initial: "foo",
+        states: {
+          foo: {},
+        },
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+
+  expect(project.extractMachines('index.ts')).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "edges": [],
+          "rootState": {
+            "data": {
+              "description": undefined,
+              "entry": [],
+              "exit": [],
+              "history": undefined,
+              "initial": "foo",
+              "invoke": [],
+              "metaEntries": [],
+              "tags": [],
+              "type": undefined,
+            },
+            "id": "",
+            "states": [
+              {
+                "data": {
+                  "description": undefined,
+                  "entry": [],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": undefined,
+                },
+                "id": "foo",
+                "states": [],
+              },
+            ],
+          },
+          "states": [],
+        },
+        [],
+      ],
+    ]
+  `);
+});
+
+test('Should extract state.initial with undefined value', async () => {
```

Comment:
I'm not sure if this is a good (future-proof) title. `initial: undefined` is so meaningless for us that I'd treat this as an error. One from which we might be able to recover but still an error.